packages/payment-processor/test/payment/erc-20-private-payment-hinkal.test.ts

```diff
@@ -0,0 +1,263 @@
+import { BigNumber, ethers, providers, Wallet } from 'ethers';
+import { RequestNetwork, Types, Utils } from '@requestnetwork/request-client.js';
+import { CurrencyTypes } from '@requestnetwork/types';
+import { EthereumPrivateKeySignatureProvider } from '@requestnetwork/epk-signature';
+import {
+  addToHinkalStore,
+  hinkalStore,
+  payPrivateErc20FeeProxyRequestFromHinkal,
+  payPrivateErc20ProxyRequestFromHinkal,
+  sendToPrivateRecipientFromHinkal,
+} from '../../src/payment/erc-20-private-payment-hinkal';
+import { getErc20Balance } from '../../src/payment/erc20';
+
+// Constants to configure the tests
+const currentNetwork: CurrencyTypes.ChainName = 'base';
+const currencyAddress = '0x833589fCD6eDb6E08f4c7C32D4f71b54bdA02913'; // USDC
+const currentCurrenyType = Types.RequestLogic.CURRENCY.ERC20;
+const currencyAmount = ethers.utils.parseUnits('0.00001', 6).toBigInt();
+const currentGateway = 'https://sepolia.gateway.request.network';
+const fee = '0';
+
+// Payer's public/private EOA keys and shielded address
+// The EOA address has a balance of 0, while the shielded address holds 1 USDC.
+const PAYER_PUBLIC_KEY = '0x44DC1e666C1ca6717849efE19eAC72AD83cFf5d2';
+const PAYER_PRIVATE_KEY = '0x3abdcb3d6d6c302a7943715d0b975ae1377d7d1d188820f6cd57b6f13fb5b0e0';
+const PAYER_SHIELDED_ADDRESS =
+  '142590100039484718476239190022599206250779986428210948946438848754146776167,0x096d6d5d8b2292aa52e57123a58fc4d5f3d66171acd895f22ce1a5b16ac51b9e,0xc025ccc6ef46399da52763a866a3a10d2eade509af27eb8411c5d251eb8cd34d';
+// PAYEE - Public EOA - Recipient of the payment
+const PAYEE = '0xA4faFa5523F63EE58aE7b56ad8EB5a344A19F266'; // some random address
+// Sender (0x01aC930fAb20482E5b7eAbF892DF57141c29de5F) - Sends funds to a private address: you will need to top up sender
+// Once the sender transfers funds to the recipient's private address, the recipient's shielded balance will be increased.
+const senderMnemonic = 'sand win seed crucial film antique adapt start pupil gallery edge collect';
+
+const RPC_URL = 'https://mainnet.base.org'; // Blockchain RPC endpoint for the Base networ
+
+jest.setTimeout(1000000); // Set Jest timeout for asynchronous operations (e.g., blockchain calls)
+
+/**
+ * Function to create a payment request for Hinkal.
+ * This sets up the Request Network with payer and payee information.
+ * @param payerWallet The wallet of the payer.
+ * @param privateKey The private key of the payer (for signing).
+ * @param type The payment network type (e.g., ERC20_PROXY_CONTRACT).
+ * @returns A Promise that resolves to the request data with events.
+ */
+export const createRequestForHinkal = async (
```

Comment:
Remove exports from test file :pray: 

```suggestion
const createRequestForHinkal = async (
```